frontend/src/lib/ParserUtils.ts

```diff
@@ -14,3 +14,20 @@ export function parseTaskDisplayName(metadata?: Metadata): string | undefined {
   }
   return taskDisplayName || componentDisplayName;
 }
+
+export function parseNodeDisplayName(nodeId: string, workflow?: Workflow): string {
```

Comment:
nit: shall we rename this to `parseTaskDisplayNameByNodeId`?
because `task` and `node` are synonyms in this context, the only difference comparing to the method above is that this gets displayName from nodeId + workflow instead of node metadata

Reply:
That makes sense, I'll rename it